rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -330,9 +338,26 @@ def _maybe_decode_simple_transactions(
                     counterparty=counterparty,
                 ))
 
+    def _maybe_decode_simple_transactions(
+            self,
+            tx: EthereumTransaction,
+            tx_receipt: EthereumTxReceipt,
+    ) -> List[HistoryBaseEntry]:
+        """Decodes normal ETH transfers, internal transactions and gas cost payments"""
+        events: List[HistoryBaseEntry] = []
+        tx_hash_hex = tx.tx_hash.hex()
+        ts_ms = ts_sec_to_ms(tx.timestamp)
+
         # check for gas spent
         direction_result = self.base.decode_direction(tx.from_address, tx.to_address)
         if direction_result is None:
```

Comment:
Why do you put `decode_internal_transaction` under `direction_result` being `None`? That's totally wrong and breaks the continuity of the previous behaviour.

All internal transactions of a transaction where an address we own is either the sender or the receiver will not be detected with what you wrote.

What you should do instead is check for internal transactions and then if we are neither a sender or receiver (`direction_result == None`) exit early

```python
	self._maybe_decode_internal_transactions(
	    tx=tx,
	    tx_receipt=tx_receipt,
	    events=events,
	    tx_hash_hex=tx_hash_hex,
	    ts_ms=ts_ms,
	)
    if direction_result is None:
        return events
```